File: backend/app/models/pack.py
from datetime import datetime

from sqlalchemy import Column, Integer, String, DateTime, Float, Boolean

from app.db.base_class import Base



class Pack(Base):
    __tablename__ = 'pack'
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    image_url = Column(String, nullable=False)
    product_quantity = Column(Integer, nullable=False)
    price = Column(Float, nullable=False)
    discount = Column(Integer, nullable=True, default=0)
    state = Column(Boolean, nullable=False, default=True)
    description = Column(String, nullable=True)
    
    created_at = Column(DateTime, nullable=False, default=datetime.utcnow())
    updated_at = Column(DateTime, nullable=False, default=datetime.utcnow())

File: itsocks-laravel/app/Models/Pack.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Pack extends Model
{
    protected $table = 'pack';

    protected $fillable = [
        'name',
        'image_url',
        'product_quantity',
        'price',
        'discount',
        'state',
        'description',
    ];

    protected $casts = [
        'price' => 'decimal:2',
        'state' => 'boolean',
        'discount' => 'integer',
        'product_quantity' => 'integer',
    ];

    public function scopeActive($query)
    {
        return $query->where('state', true);
    }
}
